drivers/adc/adc_mcux_lpadc.c

```diff
@@ -135,35 +189,72 @@ static int mcux_lpadc_start_read(const struct device *dev,
 
 	switch (sequence->oversampling) {
 	case 0:
-		data->average = kLPADC_HardwareAverageCount1;
+		tmp.hardwareAverageMode = kLPADC_HardwareAverageCount1;
 		break;
 	case 1:
-		data->average = kLPADC_HardwareAverageCount2;
+		tmp.hardwareAverageMode = kLPADC_HardwareAverageCount2;
 		break;
 	case 2:
-		data->average = kLPADC_HardwareAverageCount4;
+		tmp.hardwareAverageMode = kLPADC_HardwareAverageCount4;
 		break;
 	case 3:
-		data->average = kLPADC_HardwareAverageCount8;
+		tmp.hardwareAverageMode = kLPADC_HardwareAverageCount8;
 		break;
 	case 4:
-		data->average = kLPADC_HardwareAverageCount16;
+		tmp.hardwareAverageMode = kLPADC_HardwareAverageCount16;
 		break;
 	case 5:
-		data->average = kLPADC_HardwareAverageCount32;
+		tmp.hardwareAverageMode = kLPADC_HardwareAverageCount32;
 		break;
 	case 6:
-		data->average = kLPADC_HardwareAverageCount64;
+		tmp.hardwareAverageMode = kLPADC_HardwareAverageCount64;
 		break;
 	case 7:
-		data->average = kLPADC_HardwareAverageCount128;
+		tmp.hardwareAverageMode = kLPADC_HardwareAverageCount128;
 		break;
 	default:
 		LOG_ERR("Unsupported oversampling value %d",
 			sequence->oversampling);
 		return -ENOTSUP;
 	}
 
+	/*
+	 * Now, look at the selected channels to determine which ADC channels
+	 * we need to configure, and set those channels up.
+	 *
+	 * Since this ADC supports chaining channels in hardware, we will
+	 * start with the highest channel ID and work downwards, chaining
+	 * channels as we go.
+	 */
+	chan = CONFIG_LPADC_CHANNEL_COUNT;
+	last_enabled = 0;
+	do {
+		chan--;
+		if (sequence->channels & BIT(chan)) {
+			/* Setup this channel command */
+#if defined(FSL_FEATURE_LPADC_HAS_CMDL_MODE) \
```

Comment:
nit: this could fit in one line 80 characters

Reply:
Fixed